xds/internal/xdsclient/authority.go

```diff
@@ -399,7 +425,7 @@ func (a *authority) handleWatchTimerExpiry(rType xdsresource.Type, resourceName
 	a.resourcesMu.Lock()
 	defer a.resourcesMu.Unlock()
 
-	if state.wState == watchStateCanceled {
+	if state.wState != watchStateRespRequested {
```

Comment:
Does it make sense to throw a warning log here with the resource name/url and its watch state?

Reply:
Done!